optimum_benchmark/backends/onnxruntime.py

```diff
@@ -1,39 +1,54 @@
+from typing import Any, Callable, Dict, List, Optional, TYPE_CHECKING
+from tempfile import TemporaryDirectory
+from dataclasses import dataclass
+from logging import getLogger
+from datasets import Dataset
 import os
+
+
 import torch
 from torch import Tensor
-from datasets import Dataset
-from logging import getLogger
 from omegaconf import OmegaConf
-from dataclasses import dataclass, field
 from hydra.utils import get_class
-from tempfile import TemporaryDirectory
-from omegaconf.dictconfig import DictConfig
-from typing import Any, Callable, Dict, List, Optional
-
-
-try:
-    from onnxruntime import __version__ as onnxruntime_version
-except ImportError:
-    onnxruntime_version = "Not installed"
-
-from optimum.onnxruntime import ORTOptimizer, ORTQuantizer
+from onnxruntime import SessionOptions
+from accelerate import init_empty_weights
+from optimum.pipelines import ORT_SUPPORTED_TASKS
+from onnxruntime import __version__ as onnxruntime_version
 from optimum.onnxruntime.configuration import (
     OptimizationConfig,
     QuantizationConfig,
     AutoCalibrationConfig,
     AutoOptimizationConfig,
     AutoQuantizationConfig,
 )
+from optimum.onnxruntime import (
+    ORTOptimizer,
+    ORTQuantizer,
+    ORTTrainer,
+    ORTTrainingArguments,
+)
 
+if TYPE_CHECKING:
+    from transformers import TrainerCallback, TrainerState
+    from transformers.modeling_outputs import ModelOutput
+
+
+from .base import Backend, BackendConfig
+from .utils.optimum_utils import main_export
+from .utils.pytorch_utils import randomize_weights
+from ..profilers.ort_profiler import ORTProfilingWrapper
+from .utils.onnxruntime_utils import (
+    format_ort_quantization_dict,
+    infer_device_id,
+    DEFAULT_OPTIMIZATION_CONFIG,
+    DEFAULT_QUANTIZATION_CONFIG,
+    DEFAULT_CALIBRATION_CONFIG,
+)
 
-from optimum_benchmark.backends.base import Backend, BackendConfig
-from optimum_benchmark.backends.utils import main_export, randomize_weights
-from optimum_benchmark.profilers.ort_profiler import ORTProfilingWrapper
-from optimum_benchmark.utils import infer_device_id
 
 OmegaConf.register_new_resolver(
     "is_gpu",
-    lambda device: torch.device(device).type == "cuda",
+    lambda device: "cuda" in device.lower() or "tensorrt" in device.lower(),
```

Comment:
device is used to select the EP?

Reply:
that's a mistake, I forgot that I already decided not to go with this design in the past